### PR TITLE
JAMES-3182 Detail filter visit algorithm in method name

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`and check some more properties like being breadthFirst`

Can I argue that being breadthFirst can be witnessed by the ordering of the leaves, like `breadthFirstVisitShouldUnboxTwoLevelOperator` does it? I did split the test to enforce that property explicitly.

Your comment suggest you think of other properties to check, I could not think of others. If you have some suggestions, that would be welcome, @mbaechler .

File: server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/draft/model/FilterTest.java [hunk 1]
```diff
@@ -149,12 +169,12 @@ public void flattenShouldAllowUpToLimitNesting() {
                 throw new RuntimeException("unsuported combinaison");
```
Comment (LanKhuat) [inline server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/draft/model/FilterTest.java:169]: s/unsuported/unsupported

File: server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/draft/model/FilterTest.java [hunk 2]
```diff
@@ -166,7 +186,7 @@ public void flattenShouldRejectDeepNesting() {
...
```
Comment (LanKhuat) [inline server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/draft/model/FilterTest.java:186]: idem

**Comment by rouazana** [comment]
merged

**Comment by mbaechler** [comment]
Thanks @chibenwa , it's well tested now

**Pull request closed without merging**